aten/src/ATen/templates/Tensor.h

```diff
@@ -85,6 +86,12 @@ struct Tensor : public detail::TensorBase {
   inline Tensor toType(ScalarType t) const;
   inline Tensor toBackend(Backend b) const;
 
+  /// New-style `to()` methods.
+  Tensor to(TensorOptions options, bool non_blocking = false);
+  Tensor to(ScalarType dtype, bool non_blocking = false);
+  Tensor to(Layout layout, bool non_blocking = false);
+  Tensor to(Device device, bool non_blocking = false);
```

Comment:
Do you plan to support the `to(device, dtype=None, non_blocking=False)` signature? It is available in python. I guess that it is roughly equivalent to `to(at::Type ty, non_blocking=False)`, so maybe we can have that?

Reply:
I'm not sure I understand the one argument (not counting `non_blocking`) functions -- isn't TensorOptions all implicitly convertible from those anyway?

Reply:
@gchanan the semantics here are for e.g. `dtype()` that first `TensorOptions(*this)` (or `this->options()` when it's out) is constructed, which will copy all the options from `this`, and then *only* the `dtype` property is changed. The one taking `TensorOptions` is when you want to change *all* options.

@SsnL I guess that would be equivalent to `tensor.to(tensor.options().device(...).dtype(...))`. I think the general one here is fine